### PR TITLE
Improve handling of invalid PIV certificates

### PR DESCRIPTION
**Pull request by emlun**
Currently, both `ykman piv info` and the PIV view in ykman-qt crash if any PIV certificate slot contains data that is not a valid certificate - for example, after running `echo 'hej' | ykman piv write-object 5fc105`. These changes make the info command not crash, and allow us to also handle the same issue in the GUI.

### CODE REVIEW

**Comment by dagheyman** [comment]
Any known scenarios except `echo 'hej' | ykman piv write-object 5fc105` where this might happen?

File: test/on_yubikey/cli_piv/test_misc.py [hunk 1]
```diff
@@ -27,4 +29,19 @@ def test_write_read_object(self):
...
+                      input='Kom ihåg att du aldrig får snyta dig i mattan!')
+
+            with self.assertRaises(SystemExit):
+                ykman_cli('piv', 'export-certificate',
+                          hex(OBJ.AUTHENTICATION), '-')
+
+        def test_info_with_invalid_certificate_does_not_crash(self):
+            ykman_cli('piv', 'write-object', hex(OBJ.AUTHENTICATION), '-',
+                      '-m', DEFAULT_MANAGEMENT_KEY,
+                      input='Kom ihåg att du aldrig får snyta dig i mattan!')
```
Comment (dagheyman) [inline test/on_yubikey/cli_piv/test_misc.py]: Please use a more boring phrase, to avoid distraction. :)

**Comment by dagheyman** [comment]
Also, python2 is still supported for the library, so would be good to test it in the CI.

File: test/on_yubikey/cli_piv/test_misc.py (hunk 1 above)
Comment (emlun) [reply test/on_yubikey/cli_piv/test_misc.py]: :laughing:

**Comment by emlun** [comment]
> Any known scenarios except `echo 'hej' | ykman piv write-object 5fc105` where this might happen?

Aveen and Per ran into this somehow while working on their PIV stuff. I don't know the details, but they somehow ended up with a corrupted certificate loaded into one of the slots, at which point ykman-qt was of no help to fix it.

**Pull request merged**